### PR TITLE
ARROW-15351: [Doc][Guide] Additional tutorial for R bindings

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR adds R binding tutorial for `lubridate::mday` to the New Contributors Guide.

### CODE REVIEW

File: docs/source/developers/guide/tutorials/python_tutorial.rst [hunk 1]
```diff
@@ -133,7 +133,7 @@ Start the work on a new branch
 ------------------------------
 
 Before we start working on adding the feature we should
-create a new branch from updated master.
+create a new branch from the updated master.
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/python_tutorial.rst:136]:
```suggestion
create a new branch from the updated master branch.
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 2]
```diff
@@ -22,21 +22,429 @@
...
 
 To view the tutorial follow the
 `Walkthrough section of the Writing Bindings article <https://arrow.apache.org/docs/r/articles/developers/bindings.html#walkthrough>`_.
+
+
+R tutorial on adding a lubridate binding
+========================================
+
+In this tutorial we will document a simple binding contribution
+to Arrow R package following the steps specified by the
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:50]:
```suggestion
In this tutorial, we will document the contribution of a binding
to Arrow R package following the steps specified by the
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 3]
```diff
@@ -22,21 +22,429 @@
...
+**Let's start!**
+
+Set up
+------
+
+Let's set up the Arrow repository. We presume here that Git is
+already installed. Otherwise please see the :ref:`set-up` section.
+
+Once the `Apache Arrow repository <https://github.com/apache/arrow>`_
+is forked we will clone it and add the link of the main repository
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:77]: Do we instruct people on how to fork it?  If so, can we link to that.  If not, we should explicitly tell the reader that they'll need to fork it.

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 4]
```diff
@@ -22,21 +22,429 @@
...
+
+   $ git clone https://github.com/<your username>/arrow.git
+   $ cd arrow
+   $ git remote add upstream https://github.com/apache/arrow
+
+Building R package
+------------------
+
+Script for building R package differs depending on the Operating
+System you are using. For this reason we will only refer to
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:90]:
```suggestion
The steps to follow for for building the R package differs depending on the operating
system you are using. For this reason we will only refer to
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 5]
```diff
@@ -22,21 +22,429 @@
...
+the instructions for the building process in this tutorial.
+
+.. seealso::
+
+   For the **introduction** to the building process refer to the
+   :ref:`build-arrow-guide` section.
+
+   For the **instructions** on how to build R package refer to the
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:98]:
```suggestion
   For the **instructions** on how to build the R package refer to the
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 6]
```diff
@@ -22,21 +22,429 @@
...
+In this tutorial we will be tackling a JIRA issue for implementing
+a simple binding for ``mday()`` function that will match that of the
+existing R function from ``lubridate``. The issue can be found on
+`this link <https://issues.apache.org/jira/browse/ARROW-14816>`_.
+
+.. figure:: R_tutorial_jira_issue.jpeg
+   :scale: 40 %
+   :alt: JIRA page of the issue for the R binding.
+
+   JIRA page of the issue for the lubridate::mday binding.
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:113]:
```suggestion
   JIRA page of the issue for the lubridate::mday() binding.
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 7]
```diff
@@ -22,21 +22,429 @@
...
+       
+       Authored-by: Krisztián Szűcs <szucs.krisztian@gmail.com>
+       Signed-off-by: Sutou Kouhei <kou@clear-code.com>
+
+If we started the branch some time ago, we may need to rebase
+to upstream master to make sure there are no merge conflicts:
+
+.. code:: console
+
+   $ git pull upstream master --rebase
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:382]: Should this be `git rebase` not `git pull`?

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 8]
```diff
@@ -22,21 +22,429 @@
...
+at the code we see is not connected to our work. We can move on and
+maybe ask others if they are getting similar fail when running the tests.
+It could be we only need to rebuild the library.
+
+Check styling
+-------------
+
+As we did not change any part of the C++ code we will not
+run any linters. In case we would need to check the styling
+for the C++ part we would run ``arrow/r/lint.sh`` to lint the code.
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:272]: We still need to run `make style` in the `r` directory to style the R code

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 9]
```diff
@@ -22,21 +22,429 @@
...
+    1. testthat::expect_error(...) test-dplyr-funcs-datetime.R:187:2
+    2. testthat:::expect_condition_matching(...)
+   ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+
+   ══ Results ═════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+   Duration: 1.4 s
+
+   [ FAIL 1 | WARN 0 | SKIP 0 | PASS 230 ]
+
+There is a fail we get for the ``strftime`` function but looking
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:262]:
```suggestion
There is a failure we get for the ``strftime`` function but looking
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 10]
```diff
@@ -22,21 +22,429 @@
...
+to see how it works:
+
+.. code-block:: R
+
+   > call_function("day", Scalar$create(lubridate::ymd("2000-12-31")))
+   Scalar
+   31
+
+We can see that lubridate and Arrow functions operate on and return
+equivalent data types. Lubridate ``mday()`` function has no additional
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:176]:
```suggestion
equivalent data types. lubridate's ``mday()`` function has no additional
```

File: docs/source/developers/guide/tutorials/r_tutorial.rst [hunk 11]
```diff
@@ -22,21 +22,429 @@
...
+a Pull Request in the Apache Arrow repository.
+
+.. figure:: R_tutorial_pr.jpeg
+   :scale: 50 %
+   :alt: GitHub page of the Pull Request showing the title and a
+         description.
+
+   Here it is, our Pull Request!
+
+The Pull Request gets connected to the JIRA issue and the CI is running.
```
Comment (thisisnic) [inline docs/source/developers/guide/tutorials/r_tutorial.rst:442]:
```suggestion
The pull request gets connected to the JIRA issue and the CI is running.
```

**Review by thisisnic — COMMENTED** [review]
Just a few suggestions

File: docs/source/developers/guide/tutorials/r_tutorial.rst (hunk 3 above)
Comment (AlenkaF) [reply docs/source/developers/guide/tutorials/r_tutorial.rst:77]: We have the instructions in the guide, I will add a link to it 👍

File: docs/source/developers/guide/tutorials/r_tutorial.rst (hunk 7 above)
Comment (AlenkaF) [reply docs/source/developers/guide/tutorials/r_tutorial.rst:382]: If I understand correctly `git pull --rebase` is equal to `git fetch`& `git rebase`, just shorter (and there is something about saving you trouble when rebasing on a force-pushed rebase https://git-scm.com/book/en/v2/Git-Branching-Rebasing#_rebase_rebase_work)

File: docs/source/developers/guide/tutorials/r_tutorial.rst (hunk 8 above)
Comment (AlenkaF) [reply docs/source/developers/guide/tutorials/r_tutorial.rst:272]: Will add this, thanks for pointing it out!

File: docs/source/developers/guide/tutorials/r_tutorial.rst (hunk 7 above)
Comment (thisisnic) [reply docs/source/developers/guide/tutorials/r_tutorial.rst:399]: Oh awesome, learned something new today!

**Review by thisisnic — APPROVED** [review]
Looks great

**Pull request closed without merging**